### PR TITLE
5-get-articles

### PR DESCRIPTION
**Pull request by HaylzRandom**
Completed section 5 of getting all articles from API and ordering them by default in descending order by created_at. 

Added option to change whether the order is ASC or DESC.

Refactored error handling into own page so cleaner code and more abstracted.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -110,3 +111,73 @@ describe('/api/articles/:article_id', () => {
...
+        - respond with 400 when passed an invalid query
+  */
+  test('GET:200 should send an array of all articles', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+
+        expect(articles).toHaveLength(5);
```
Comment (two-triangles) [inline __tests__/app.test.js]: Should we not be getting back 13 articles?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -110,3 +111,73 @@ describe('/api/articles/:article_id', () => {
...
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+
+        expect(articles).toBeSortedBy('created_at', {
+          descending: true,
+        });
+      });
+  });
+  test('GET:200 should send array sorted by date in ascending order when passed as a query', () => {
```
Comment (two-triangles) [inline __tests__/app.test.js:156]: This task isn't meant to be done till later! Well done none the less but please make sure you are reading the tasks clearly.

File: app.js [hunk 3]
```diff
@@ -11,32 +19,19 @@ app.use(express.json());
...
-        res.status(500).send({ msg: 'Internal Server Error' });
-    }
-  }
-
-  if (err.status) {
-    res.status(err.status).send({ msg: err.msg });
-  }
-});
+// Error Handling
+app.use(handleCustomErrors);
```
Comment (two-triangles) [inline app.js:33]: Lovely refactor here :)

File: __tests__/app.test.js (hunk 1 above)
Comment (two-triangles) [reply __tests__/app.test.js]: I am wondering if this due to how you have joined your table. You are joining on where article.article_id = comment.article_id - what about articles that have no comments? They are currently going to be missed out.

File: __tests__/app.test.js (hunk 2 above)
Comment (HaylzRandom) [reply __tests__/app.test.js:156]: Apologies! I figured as we were defaulting the sort order it was logical :)

**Comment by two-triangles** [comment]
Looks great Haley, happy for you to merge into your main branch :)

**Pull request merged**